tensorboard/components/tensor_widget/tensor-widget-impl.ts

```diff
@@ -526,18 +543,35 @@ export class TensorWidgetImpl implements TensorWidget {
           this.renderSelection();
         });
         valueDiv.addEventListener('mouseenter', () => {
-          const detailedValue = valueDiv.getAttribute(DETAILED_VALUE_ATTR_KEY);
-          if (!detailedValue) {
+          // The DETAILED_VALUE_ATTR_KEY attribute of the value div holds a
+          // "detailed" string representation of the value of the corresponding
+          // tensor element. It is detailed in the sense that it has as many
+          // decimal points as supported by JavaScript's string representation
+          // of numbers, in the case of float dtypes. The only exceptions
+          // are very long string elements, which we still truncate in order
+          // to avoid overtaxing the DOM.
+          const valueAttr = valueDiv.getAttribute(DETAILED_VALUE_ATTR_KEY);
+          if (!valueAttr) {
             return;
           }
+          let detailedValue: number | boolean | string = JSON.parse(valueAttr);
+          const valueClass = this.getValueClass();
+          if (valueClass !== 'string' && !detailedValue) {
+            return;
+          }
+
+          if (valueClass === 'boolean') {
```

Comment:
Pretty sure this one is now wrong too.  If `detailedValue === true`, then `detailedValue !== '1'`, and so it gets converted to `false`.  Can this be deleted, or can you instead use it to check that `detailedValue` is already  boolean?

Reply:
Done.